unstructured/ingest/connector/kafka.py

```diff
@@ -0,0 +1,315 @@
+        from confluent_kafka import Producer
+
+        is_confluent = self.connector_config.confluent
+        bootstrap = self.connector_config.bootstrap_server
+        port = self.connector_config.port
+
+        conf = {
+            "bootstrap.servers": f'{bootstrap}:{port}',
+            "client.id": socket.gethostname(),
+        }
+
+        if is_confluent:
+            api_key = self.connector_config.access_config.kafka_api_key
+            secret = self.connector_config.access_config.secret
+            conf["sasl.mechanism"] = "PLAIN"
+            conf["security.protocol"] = "SASL_SSL"
+            conf["sasl.username"] = api_key
+            conf["sasl.password"] = secret
+
+        producer = Producer(conf)
+        logger.debug(f"Connected to bootstrap: {bootstrap}")
+        return producer
+
+    @DestinationConnectionError.wrap
+    def check_connection(self):
+        _ = self.kafka_producer
+
+    @staticmethod
+    def chunks(iterable, batch_size=100):
+        """A helper function to break an iterable into chunks of size batch_size."""
+        it = iter(iterable)
+        chunk = tuple(itertools.islice(it, batch_size))
+        while chunk:
+            yield chunk
+            chunk = tuple(itertools.islice(it, batch_size))
+
+    @DestinationConnectionError.wrap
+    def upload_msg(self, batch) -> int:
+        logger.debug(f"Uploding batch: {batch}")
+        topic = self.connector_config.topic
+        producer = self.kafka_producer
+        uploaded = 0
+        for i in range(len(batch)):
+            filename = f'{batch[i].pop("filename")}'
+            producer.produce(topic, key=filename, value=str(batch[i]))
+            uploaded += 1
+        return uploaded
+
+    @DestinationConnectionError.wrap
+    def write_dict(self, *args, dict_list: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+        logger.info(f"Writing {len(dict_list)} documents to Kafka")
+        BATCH_SIZE = 4
+        num_uploaded = 0
+
+        """
+        if self.write_config.num_processes == 1:
+           #self.upload_msg("asdf")
+           for chunk in self.chunks(dict_list, BATCH_SIZE):
+                num_uploaded += self.upload_msg(chunk)  # noqa: E203
+        else:
+            with mp.Pool(processes=self.write_config.num_processes) as pool:
+                pool.map(self.upload_msg, list(self.chunks(dict_list, BATCH_SIZE)))
+        """
+        for chunk in self.chunks(dict_list, BATCH_SIZE):
+            num_uploaded += self.upload_msg(chunk)  # noqa: E203
+
+        producer = self.kafka_producer
+        producer.flush()
+        logger.info(f"Uploaded {num_uploaded} documents to Kafka")
+
+    def write(self, docs: t.List[BaseIngestDoc]) -> None:
+        content_list: t.List[t.Dict[str, t.Any]] = []
+        for doc in docs:
+            local_path = doc._output_filename
+            with open(local_path) as json_file:
+                dict_content = json.load(json_file)
+                for content in dict_content:
```

Comment:
I notice that we are only writing `type, text, filename` into kafka. Why are we limiting ourself to this? Is this a limit of Kafka? 
Can we pass in embeddings and the other metadata? 

Maybe we should skip the chunking and embedding phase then in the destination test...
Please add some info in the PR description about the limitations or exact purpose of the connectors.